### ISSUE TITLE
Does LibVMI freeze all cores of the VM when a memory hook gets hit? (Xen)

### ISSUE THREAD

**Issue by subvert0r** [libvmi/libvmi#1104]
Assume I have used `vmi_register_event` to register a hook on a windows kernel API on a guest VM that has multiple core.

My question is, when my hook gets hit and my callback gets called, will LibVMI freeze the execution all cores of the guest windows VM? Or only the core that the hook was hit?

**Comment by tklengyel** [comment]
Depends on the underlying hypervisor implementation but Xen/KVM does not freeze the whole VM, only the core that trapped to the hypervisor for the event. You can then of course in your callback request the other cores to be paused as well but you should not expect these cores will get paused at the same time as the first one did.

**Comment by tklengyel** [comment]
DRAKVUF doesn't pause the whole VM either for processing callbacks.

**Comment by tklengyel** [comment]
Not necessarily, depends on what you are trapping on. It's certainly a bottle-neck that the event interface is serialized FIFO and could be improved. How much that improvement will mean will largely depend on what your application is monitoring and what the monitored VM is doing.

**Comment by rjadidi920** [comment]
> Not necessarily, depends on what you are trapping on. It's certainly a bottle-neck that the event interface is serialized FIFO and could be improved. How much that improvement will mean will largely depend on what your application is monitoring and what the monitored VM is doing.

What do you mean by depends on  what I am trapping on? Do you mean it depends on the function that I am hooking or..? If so, why?
